sched/semaphore/sem_holder.c

```diff
@@ -1035,18 +1014,57 @@ void nxsem_release_holder(FAR sem_t *sem)
 {
   FAR struct tcb_s *rtcb = this_task();
   FAR struct semholder_s *pholder;
+  FAR struct semholder_s *candidate = NULL;
+  unsigned int total = 0;
 
   /* Find the container for this holder */
 
-  pholder = nxsem_findholder(sem, rtcb);
-  if (pholder != NULL && pholder->counts > 0)
+#if CONFIG_SEM_PREALLOCHOLDERS > 0
+  for (pholder = sem->hhead; pholder; pholder = pholder->flink)
```

Comment:
```suggestion
  for (pholder = sem->hhead;
       pholder != NULL && pholder->counts > 0;
       pholder = pholder->flink)
```

Reply:
Holder's count does not need to be checked because it will be freed in nxsem_restore_baseprio if it's count is 0.

Reply:
Holder's count does not need to be checked because the holder will be freed in nxsem_restore_baseprio if it's count is 0.